modules/programs/powerline-go.nix

```diff
@@ -0,0 +1,120 @@
+    else if builtins.isAttrs value then
+      valueToString
+      (mapAttrsToList (key: val: "${valueToString key}=${valueToString val}")
+        value)
+    else
+      builtins.toString value;
+
+  modulesArgument = optionalString (cfg.modules != null)
+    "-modules ${valueToString cfg.modules}";
+
+  newlineArgument = optionalString cfg.newline "-newline";
+
+  pathAliasesArgument = optionalString (cfg.pathAliases != null)
+    "-path-aliases ${valueToString cfg.pathAliases}";
+
+  otherSettingPairArgument = name: value:
+    if value == true then "-${name}" else "-${name} ${valueToString value}";
+
+  otherSettingsArgument = optionalString (cfg.settings != { })
+    (concatStringsSep " "
+      (mapAttrsToList otherSettingPairArgument cfg.settings));
+
+  commandLineArguments = ''
+    ${modulesArgument} ${newlineArgument} ${pathAliasesArgument} ${otherSettingsArgument}
+  '';
+
+in {
+  meta.maintainers = [ maintainers.DamienCassou ];
+
+  options = {
+    programs.powerline-go = {
+      enable = mkEnableOption "Powerline-go";
+
+      modules = mkOption {
+        default = null;
+        type = types.nullOr (types.listOf types.str);
+        description = ''
+
+          List of module names to load. The list of all available
+          modules as well as the choice of default ones are at <link
+          xlink:href="https://github.com/justjanne/powerline-go">https://github.com/justjanne/powerline-go</link>.
+
+        '';
+        example = [ "host" "ssh" "cwd" "gitlite" "jobs" "exit" ];
+      };
+
+      newline = mkOption {
+        default = false;
+        type = types.bool;
+        description =
+          "Set to true if the prompt should be on a line of its own.";
+        example = true;
+      };
+
+      pathAliases = mkOption {
+        default = null;
+        type = types.nullOr (types.attrsOf types.str);
+        description = ''
+          Pairs of full-path and corresponding desired short name. You
+          may use '~' to represent your home directory but you should
+          protect it to avoid shell substitution.
+        '';
+        example = { "\\~/projects/home-manager" = "prj:home-manager"; };
+      };
+
+      settings = mkOption {
+        default = { };
+        type = with types; attrsOf (oneOf [ bool int str (listOf str) ]);
+        description = ''
+          This can be any key/value pair as described in <link
+          xlink:href="https://github.com/justjanne/powerline-go">https://github.com/justjanne/powerline-go</link>.'';
+
+        example = literalExample ''
+          {
+            hostname-only-if-ssh = true;
+            numeric-exit-codes = true;
+            cwd-max-depth = 7;
+            ignore-repos = [ "/home/me/big-project" "/home/me/huge-project" ];
+          }
+        '';
+      };
+
+      extraUpdatePS1 = mkOption {
+        default = "";
+        description =
+          "Shell code to add at the end of the _update_ps1() function.";
```

Comment:
Never used powerline. Is the _update_ps1() funtcion documented in the powerline docs? If not, I think it should be a longer decsription.

Reply:
Ah, I see. It's the command setting the prompt?. Mabye give a hint that this sets the prompt?

I'm not sure if we should write something here, of If I am the problem, and those who use this option do know the PS1 function.

Reply:
The `_update_ps1()` function is indeed [part of powerline-go](https://github.com/justjanne/powerline-go#bash). I changed the description anyway to make it clearer.